GHCJS/Foreign.hs

```diff
@@ -481,6 +481,89 @@ unsafeGetPropMaybe p o = do
   if isUndefined p' then return Nothing else return (Just p')
 {-# INLINE unsafeGetPropMaybe #-}
 
+{- | Read a property chain from a JS object. Throws a 'JSException' if
+     o is not a JS object or one of the properties in chain
+     cannot be accessed. Useful for reading nested properties, e.g. to
+     read value from `window.location.href` use
+     '["location","href"] `getPropCascade` windowObj'
+-}
+getPropCascade :: ToJSString a
```

Comment:
Type constrain should have `b ~ c`, which requires **TypeFamilies** language extension.